libraries/bezier-rs/src/subpath/lookup.rs

```diff
@@ -10,6 +12,70 @@ impl Subpath {
 		self.iter().fold(0., |accumulator, bezier| accumulator + bezier.length(num_subdivisions))
 	}
 
+	fn global_euclidean_to_local_euclidean(&self, global_t: f64) -> (usize, f64) {
+		let lengths = self.iter().map(|bezier| bezier.length(None)).collect::<Vec<f64>>();
+		let total_length: f64 = lengths.iter().sum();
+
+		let mut accumulator = 0.;
+		for (index, length) in lengths.iter().enumerate() {
+			let length_ratio = length / total_length;
+			if accumulator <= global_t && global_t <= accumulator + length_ratio {
+				return (index, (global_t - accumulator) / length_ratio);
+			}
+			accumulator += length_ratio;
+		}
+		(0, 0.)
+	}
+
+	/// Convert a [SubpathTValue] to a parametric `(segment_index, t)` tuple.
+	/// - Asserts that `t` values contained within the `SubpathTValue` argument lie between 0 and 1.
+	/// - If the argument is a variant containing a `segment_index`, asserts that the index references a valid segment on the curve.
+	pub(crate) fn t_value_to_parametric(&self, t: SubpathTValue) -> (usize, f64) {
```

Comment:
This should probably return an `Option` so we don't panic if you call it with invalid inputs. That means changing your `assert`s and `unwrap`s to return `None`.

Reply:
We included the same assert logic in the `Bezier` version of `t_value_to_parametric` as well. We put in these assertions since otherwise we'd either need to place these assertions in every function that requires a tvalue, or have all those functions return optionals, which we thought cluttered the interface somewhat.

I'd like to keep the bezier and subpath versions consistent, so removing the assertions and converting everything to use optionals will be a more significant change. @Keavon should we do this as part of this MR or a separate one?

Reply:
I hadn't considered how every function would have to return an Option, good point. Let's defer on this.